app/controllers/chats_controller.rb

```diff
@@ -0,0 +1,101 @@
+class ChatsController < ApplicationController
+	before_action :authenticate_user!
+	def index
+		all_recipients
+
+		if @notified_users.count > 0
+			@user = @notified_users.first
+		elsif @followed_users.count > 0
+			@user = @followed_users.first
+		else
+			@user = @all_users.first
+		end
+		@msgs = get_messages(@user)
+		get_read(@msgs)
+	end
+
+	def msg
+		all_recipients
+		
+		@user = User.find(params[:id])
+		@msgs = get_messages(@user)
+		get_read(@msgs)
+	end
+
+	def send_msg
+
+		@user = User.find(params[:chat][:receiver_id])
+		chat = Chat.new(chat_params)
+		chat.sender_id = current_user.id
+		chat.message_status = "unread"
+		
+		if params[:chat][:designs_id].present?
+			chat.message_type = "image"
+			chat.designs_id = params[:chat][:designs_id].split
+		else
+			chat.message_type = "text"
+		end
+
+		if (FollowingList.blocked.where(from_id: params[:chat][:receiver_id], to_id: current_user).count == 0) && (params[:chat][:body] != "" || params[:chat][:designs_id] != "")
+			chat.save!
+		end
+		@msgs = get_messages(@user)
+	end
+
+	def search
+		if params[:name].blank?
+			@search_users = []
+		else
+			all = User.where("name LIKE ?","%#{params[:name]}%").pluck(:id).sort - [current_user.id]
```

Comment:
move this logic from controller and make class method or scope which returns expected result. never write such queries in controller. It is not controller's part. Move them to model and make method call here.

Reply:
calling #pluck after each query is suspicious to me. I would like to know why we are doing this? I see this many times in the previous code